### PR TITLE
Rubocop is your friend part 6

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
@openSUSE/open-build-service Please review

**Comment by bgeuken** [comment]
> Ana06 reviewed
> Is df7a48d corrected manually? :thinking: can you please mention in the commit message if the things were or not autocorrected?

I do, as you can see in the other commits.

> I love to see Rails/Present and Rails/Blank offenses solved :green_heart:

:tada:

**Comment by Ana06** [comment]
> > Ana06 reviewed
> > Is df7a48d corrected manually?
> 
> No, it wasnt'.

Are you sure? because I think `or` and `||` doesn't have the same priority and Rubocop shouldn't correct them 🤔

**Comment by bgeuken** [comment]
> Ana06 reviewed
> Are you sure? because I think or and \|\| doesn't have the same priority and Rubocop shouldn't correct them

@Ana06 
 I am. The cop is also makred as [auto-correctable](https://github.com/openSUSE/open-build-service/pull/4233/files#diff-74423563ec7c74f4cb7f74689b49e7c1L407).

Either that auto-correct just corrects the simple cases or they consider the priority. Given that the cop added brackets, I would think it's the latter.

That said, the code is crazy. I am fixing it.

File: src/api/app/views/message/_messages.xml.builder [hunk 1]
```diff
@@ -8,7 +6,7 @@ root_attrs[:package] = @package.name if @package
 xml.messages(root_attrs) do
   @messages.each do |msg|
     attrs = {}
-    attrs[:severity] = msg.severity or attrs[:severity] = 0
+    attrs[:severity] = msg.severity || 0
```
Comment (bgeuken) [inline src/api/app/views/message/_messages.xml.builder]: @DavidKang Please double check^^

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -546,7 +546,7 @@ def changestate_accepted(opts)
   end
 
   def changestate_revoked
-    bs_request_actions.where(type: 'maintenance_release').each do |action|
+    bs_request_actions.where(type: 'maintenance_release').find_each do |action|
```
Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: [nitpick] The where shouldn't be necessary!

File: src/api/app/models/event/base.rb [hunk 3]
```diff
@@ -287,7 +287,7 @@ def _roles(role, project, package = nil)
     end
 
     def send_to_bus
-      RabbitmqBus.publish(self.class.message_bus_routing_key, read_attribute(:payload))
+      RabbitmqBus.publish(self.class.message_bus_routing_key, self[:payload])
```
Comment (ChrisBr) [inline src/api/app/models/event/base.rb]: Why is ``self[:payload]`` here necessary? Just ``payload`` should work as well, no?

Comment (mdeniz) [reply src/api/app/models/event/base.rb]: In general is better to use `self` if you want to be sure that you are setting the attribute and not a method that overrides what ActiveRecord defines.

**Comment by DavidKang** [comment]
Superseded by: #4237, because the commit [13d6209](https://github.com/openSUSE/open-build-service/pull/4233/commits/13d62090616422a45ed554e5bd5ca0ca8d03eca0) is causing some test failures and to not block the others commits, we will fix it in another PR

**Comment by Ana06** [comment]
@bgeuken that the cop supports autocorrect doesn't mean that all the offenses are safety autocorrected and Rubocop correct only the safes ones.

**Pull request closed without merging**